python/cuml/linear_model/linear_regression.pyx

```diff
@@ -303,8 +309,14 @@ class LinearRegression(LinearPredictMixin,
 
         """
         cdef uintptr_t X_ptr, y_ptr, sample_weight_ptr
+
+        need_explicit_copy = self.copy_X and hasattr(X, "__cuda_array_interface__") and X.shape[1] == 1
```

Comment:
The more I think about it the more I think we should do the default of `copy_X` to a value like `auto`, with `True` _always_ making a copy and `False` _never_ making a copy. The reasoning behind this deviation from sklearn, is that we want to preserve GPU memory as much as possible by default, otherwise at least we need to chenge the docstring description to reflect that we do a copy _only in this case_ at least.

Reply:
How do you want to deal with the numerous cases where we have to make a copy and the argument is `False`? Raise an exception? I think the here implemented approach, where `copy_X` means that we guarantee to make a copy whereas `False` means we might make a copy is sensible and not a foreign concept to users. I am not rejecting your proposal, I just want to make sure we think it through.

The risk I see with the implemented approach is that the default value is True which is "safer", but poses a performance regression for some cases. Changing it to `False` avoids that, but bears the risk of users experiencing problems as described in the addressed issue.

Reply:
From offline discussion: We will stick to the implementation, but add an explicit note to the docs and log entry about the minor change in behavior.

Reply:
Addressed in https://github.com/rapidsai/cuml/pull/5495/commits/7a7ffc2687a99f53537076607a2405c10b145158 .